sql/functions/create_partition_time.sql

```diff
@@ -272,6 +291,7 @@ FOREACH v_time IN ARRAY p_partition_times LOOP
         SELECT
             sub_parent
             , sub_control
+            , sub_time_encoder
```

Comment:
Doesn't subpartitioning also need the decoder function as well?

Reply:
Good catch! I missed one use case completely to allow sub partitioning with text columns. Have fixed this and also included a test for this